packages/casting/src/follower-cosmjs.js

```
@@ -217,14 +221,13 @@ export const makeCosmjsFollower = (
   };
 
   /**
+   * @deprecated no longer supported https://github.com/cosmos/cosmjs/pull/1623
    * @param {number} [height]
    * @returns {Promise<QueryStoreResponse>}
    */
   const getProvenDataAtHeight = async height => {
-    return retryGetPrefixedData(async (endpoint, storeName, storeSubkey) => {
-      const queryClient = await provideQueryClient(endpoint);
-      return E(queryClient).queryStoreVerified(storeName, storeSubkey, height);
-    });
+    console.error('getProvenDataAtHeight', height, 'is no longer supported');
+    throw makeError(X`Verified queries are no longer supported`);
```

This doesn't break anything? I'd expect at least that the old default "optimistic" starts showing a lot of errors/warnings, but I'm not sure where, and tests won't exactly catch noisy logs. Maybe we could make a temporary change to fail earlier if the proof is "optimistic" so we can see if any tests fail? Then we'd know which code is impacted.

---

It's marked as a breaking change, but no CI tests were affected. I don't think anyone was opting into optimistic or strict, but if they were there's not really anything more we can do. There's no implementation anymore.

In this case I think it's best to fail loudly so any code expecting verification fails and can be updated.

---

> In this case I think it's best to fail loudly so any code expecting verification fails and can be updated.

Right, this will fail loudly for `strict`, but for `optimistic` which is the default, the follower will yield the data and then throw later when this proof verification fails. I don't think this would "break" anything, but result in some noisy logs saying the proofs failed. I was thinking it might be better to throw earlier, so that we can catch those cases, but if no one was using optimistic then that's moot.

---

That would be a problem but I'm pretty sure this PR changes the default to 'none'.